lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -207,3 +261,27 @@ func loadAPNs(gateway *protos.Identity_Gateway) (map[string]*lte_models.ApnConfi
 
 	return apnsByName, apnResources, nil
 }
+
+// injectAPNResources adds the gateway-specific apn resources data to subscriber
+// protos before returning to AGWs.
+func injectAPNResources(subProtos []*lte_protos.SubscriberData, gateway *protos.Identity_Gateway) ([]*lte_protos.SubscriberData, error) {
+	_, apnResources, err := loadAPNs(gateway)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range subProtos {
+		subProto := subProtos[i]
+		if subProto.GetNon_3Gpp().GetApnConfig() == nil {
+			continue
+		}
+		for j := range subProto.Non_3Gpp.ApnConfig {
+			apnKey := subProto.Non_3Gpp.ApnConfig[j].ServiceSelection
+			if apnResourceModel, ok := apnResources[apnKey]; ok {
+				subProto.Non_3Gpp.ApnConfig[j].Resource = apnResourceModel.ToProto()
+			}
+		}
+		subProtos[i] = subProto
```

Comment:
Reading and setting subProto like this is idiosyncratic. Try "for i, subProto := range subProtos", then just edit the subProto in-place (it's a ptr, so no need to add it back to the array)